plugins/sf2_player/sf2_player.cpp

```diff
@@ -153,14 +153,12 @@ sf2Instrument::sf2Instrument( InstrumentTrack * _instrument_track ) :
 	m_chorusDepth.setInitValue(settingVal);
 #endif
 
-	loadFile( ConfigManager::inst()->sf2File() );
-
-	updateSampleRate();
-	updateReverbOn();
-	updateReverb();
-	updateChorusOn();
-	updateChorus();
-	updateGain();
+	// FIXME: there's no good way to tell if we're loading a preset or an empty instrument
+	// We rely on instantiate() to load the default soundfont for new instruments,
+	// but we don't need that when loading a project/preset/preview
+	loadFile(Engine::getSong()->isLoadingProject() or instrumentTrack()->isPreviewMode()
+				? ""
+				: ConfigManager::inst()->sf2File());
```

Comment:
We prefer to use `||` over the `or` keyword. Also, it looks like `loadFile` won't do anything if passed an empty string, so perhaps it makes more sense not to call it at all? (It won't call `openFile` if the string is empty, and if no soundfont is loaded, the bank/patch selection loop won't run.)
```suggestion
	if (!Engine::getSong()->isLoadingProject() && !instrumentTrack()->isPreviewMode())
	{
		loadFile(ConfigManager::inst()->sf2File());
	}
```

Reply:
> This is fine if fluidsynth's default value matches the model's default value, but ideally we wouldn't rely on that.

Good observation. This seems to be the case with `updateChorus` and `updateReverb` too, [when running Fluid version 1](https://github.com/LMMS/lmms/blob/ddff2eba9f430ea0cdde087ba4f08fdfa329d936/plugins/sf2_player/sf2_player.cpp#L133). But as you say, as long as everything gets updated inside `updateSampleRate` this is perfectly fine.

> I think we should rename updateSampleRate at some point

Why not do it right away? It's only used in two places. And this PR makes things more confusing by relying on it for everything. What would be a good name? `reloadSynth` ?